pandas/core/dtypes/concat.py

```diff
@@ -173,6 +174,10 @@ def is_nonempty(x):
     elif 'sparse' in typs:
         return _concat_sparse(to_concat, axis=axis, typs=typs)
 
+    extensions = [is_extension_array_dtype(x) for x in to_concat]
+    if any(extensions) and not all(extensions):
```

Comment:
do you have an actual case where the "and not all(extensions)" is hit? 
Because if that is the case, I also think the `np.concatenate(to_concat, axis=axis)` will fail as it has the wrong dimensions to both work at the same time?

Reply:
Uhmm I suppose I don't have an example, because the all extension dtype case should never get here, right? It should be going down the `is_uniform_join_units` path. I'll change it, to remove the `all(extensions)` condition, so that it makes sense locally.